src/client/datascience/jupyter/kernels/kernelSelections.ts

```diff
@@ -307,6 +310,15 @@ export class KernelSelectionProvider {
         LiveKernelConnectionMetadata | KernelSpecConnectionMetadata
```

Comment:
Perhaps there's too many classes in this file? Maybe a separate directory and have each class in its own file?

Reply:
Will do

Reply:
Looking at it now, adding small classes into a single file was never a good idea, they tend to grow, i.e. lesson learnt, always put them into separate files.

Reply:
Fixed